libsolidity/analysis/TypeChecker.cpp

```diff
@@ -453,7 +453,12 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 		if (contractType->contractDefinition().isLibrary())
 			m_errorReporter.typeError(1273_error, _variable.location(), "The type of a variable cannot be a library.");
 	if (_variable.value())
-		expectType(*_variable.value(), *varType);
+	{
+		if (dynamic_cast<MappingType const*>(varType))
+			m_errorReporter.typeError(6280_error, _variable.location(), "Mappings cannot be assigned to.");
```

Comment:
This should only be an error for state variables.

Reply:
For local variables, `_variable.value()` would be `nullptr` I think. So is there a case that I'm missing?

Reply:
Ah you are right! Can you add an explicit check nevertheless?

Reply:
Sure.